Logger/FileLogger.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.IO;
+
+namespace Logger
+{
+    public class FileLogger : BaseLogger
+    {
+        private string? filePath;
+        public string FilePath
+        {
+            get => filePath!;
+            set => filePath = value ?? throw new ArgumentNullException(null);
+        }
+
+        public FileLogger(string filePath)
+        {
+            FilePath = filePath;
+        }
+
+        public override void Log(LogLevel logLevel, string message)
+        {
+            if (FilePath == null || !File.Exists(FilePath))
```

Comment:
```suggestion
            if (FilePath == null)
```
StreamWriter will programmatically create your file for you - so you don't need this code. This line is part of what is causing one of your unit tests to fail.